### ISSUE TITLE
sriov-network-operator should not depend on hardcoded supported device list

### ISSUE THREAD

**Issue by adrianchiris** [k8snetworkplumbingwg/sriov-network-operator#86]
The `NicIdMap` is used for generating device plugin configmap and/or perhaps in other code path (operator-webhook).
If we can find a way to remove all those direct uses of NicIdMap, we could delete it and allows more NICs to be used with SR-IOV Operator directly.

For platforms that wish to limit the devices which the sriov-network-operator operates on, a configMap may be used to provide the list of supported devices.

**Comment by pliurh** [comment]
Agree. As OLM has supported ConfigMap https://github.com/operator-framework/operator-lifecycle-manager/issues/1120.
